cfme/common/provider.py

```diff
@@ -707,18 +645,24 @@ def load_all_provider_templates(self):
             return True
 
 
-@fill.method((Form, BaseProvider.Credential))
+@fill.method((Form, cfme.base.credential.Credential))  # default credential
+@fill.method((Form, cfme.base.credential.AMQPCredential))
+@fill.method((Form, cfme.base.credential.CANDUCredential))
+@fill.method((Form, cfme.base.credential.AzureCredential))
+@fill.method((Form, cfme.base.credential.SSHCredential))
+@fill.method((Form, cfme.base.credential.TokenCredential))
+@fill.method((Form, cfme.base.credential.ServiceAccountCredential))
```

Comment:
**REQUIRED**  given the changed structure, it should either keep using only the base class or completely split the lookup